src/cuda/CUDACore/cudaCheck.h

```diff
@@ -6,6 +6,10 @@
 #include <sstream>
 #include <stdexcept>
 
+// Boost headers
+#define BOOST_STACKTRACE_USE_BACKTRACE
```

Comment:
`boost/stacktrace.h` can be used without libbacktrace if one does not set `-DBOOST_STACKTRACE_USE_BACKTRACE`, but the result is less useful; in particular it does not use debug symbols to display line numbers.

If you prefer to make libbacktrace optional, then we can move the flag to the Makefile (but I'm not sure where).
If we can always have libbacktrace, then keeping the symbol in the source seems the simplest approach.

Reply:
Can another file including `boost/stacktrace.hpp` without `#define BOOST_STACKTRACE_USE_BACKTRACE` cause problems in the same program?

On the other hand, given that the purpose of this project is prototyping and testing (in contrast to production and huge number of files), maybe having the `#define` here is the most cost-effective way.

Reply:
> Can another file including `boost/stacktrace.hpp` without `#define BOOST_STACKTRACE_USE_BACKTRACE` cause problems in the same program?

Yes, it would be an ODR violation.

> On the other hand, given that the purpose of this project is prototyping and testing (in contrast to production and huge number of files), maybe having the `#define` here is the most cost-effective way.

The alternative is to move this flag to the `Makefile`, under the Boost section - but if I do that, then I should add `libbacktrace` as a dependency to any program that uses Boost ?

Reply:
I think I'm leaning towards keeping the `#define` here. In case another use appears within the same program, we can revisit the choice.